app/models/vm/operations.rb

```diff
@@ -14,7 +14,11 @@ module Vm::Operations
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s
+    if MiqServer.active_miq_servers().where(:has_active_cockpit_ws => true)
+        "/dashboard/cockpit_redirect?" + "ip=" + ipaddress
+    else
+        URI::HTTP.build(:host => ipaddress, :port => 9090)
```

Comment:
I think this line should be
`URI::HTTP.build(:host => ipaddresses.first, :port => 9090).to_s`
￼